src/main/java/org/terracotta/statistics/TableSkeleton.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.statistics;
+
+import org.terracotta.statistics.extended.StatisticType;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.terracotta.statistics.ConstantValueStatistic.instance;
+
+/**
+ * @author Mathieu Carbou
+ */
+public abstract class TableSkeleton implements Serializable {
```

Comment:
This abstract class seems odd... and why is there no interface defining the contract of a table?

Reply:
Because I think of a table as being the output of a Statistic, like we could have Long, Double, String, Table, etc. All these types are serializables and just holds some data there is no specific contract, like there is no interface on top of the `Number` class